pkg/chunk/fifo_cache.go

```diff
@@ -106,8 +183,17 @@ func (c *fifoCache) get(key string) (value interface{}, updated time.Time, ok bo
 	var index int
 	index, ok = c.index[key]
 	if ok {
-		value = c.entries[index].value
-		updated = c.entries[index].updated
+		updated := c.entries[index].updated
+		if time.Now().Sub(updated) < c.validity {
+			value = c.entries[index].value
+			return
+		}
+
+		c.metrics.totalMisses.Inc()
+		c.metrics.staleGets.Inc()
+		ok = false
+	} else {
+		c.metrics.totalMisses.Inc()
```

Comment:
Try to un-nest this a little more:

```go
index, ok = c.index[key]
if !ok {
  c.metrics.totalMisses.Inc()
  return
}

updated := c.entries[index].updated
if time.Now().Sub(updated) > c.validity {
  c.metrics.totalMisses.Inc()
  c.metrics.staleGets.Inc()
  ok = false
  return
}

value = c.entries[index].value
return
}
```

Also might be worth getting rid of the named return values at this point.